flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/AdaptiveBatchScheduler.java

```diff
@@ -179,6 +183,7 @@ protected void onTaskFinished(final Execution execution, final IOMetrics ioMetri
         checkNotNull(ioMetrics);
         updateResultPartitionBytesMetrics(ioMetrics.getResultPartitionBytes());
         initializeVerticesIfPossible();
+        tryEnrichInputNumBytesForExecutionVertexInputInfos();
```

Comment:
How about calculating the `InputNumBytes` when it 's really needed? We can calculate it in `ExecutionVertex#getExecutionVertexInputNumBytes`, and then we don't need the `setAggregatedInputNumBytes` and `getAggregatedInputNumBytes`